urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/price/PriceFactory.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * PriceFactory.java
+ */
+package price;
+
+import exceptions.ExceptionHandler;
+import java.util.HashMap;
+
+/**
+ * This class creates Price objects for the DePaul Stock Exchange Project.<br>
+ * This class implements a simple factory/flyweight design pattern.
+ *
+ * @author Kevin Patrick Westropp
+ * @version StockExVirtuoso.Version.1.0
+ */
+public class PriceFactory {
+
+    /**
+     * HashMap which contains all prices values made so far.
+     */
+    private static HashMap<Integer, Price> prices = new HashMap<>();
+
+    /**
+     * Public Factory method takes an incoming String value and converts it to a
+     * long value then passes it to the makeLimitPrice(long value) to return a
+     * new Price object.
+     *
+     * @param value of new price object
+     * @return Limit Price object
+     * @throws Exception
+     * @see #makeLimitPrice(long)
+     */
+    public static Price makeLimitPrice(String value) throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 23. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.